prow/config/config_test.go

```diff
@@ -1148,6 +1148,94 @@ func TestValidateLabels(t *testing.T) {
 	}
 }
 
+func TestValidateMultipleContainers(t *testing.T) {
+	ka := string(prowjobv1.KubernetesAgent)
+	goodSpec := v1.PodSpec{
```

Comment:
Please include test cases for decorated jobs. This podspec wouldn't be valid for a decorated job.